src/modules/Game.class.js

```diff
@@ -6,63 +6,295 @@
  * Feel free to add more props and methods if needed.
  */
 class Game {
-  /**
-   * Creates a new game instance.
-   *
-   * @param {number[][]} initialState
-   * The initial state of the board.
-   * @default
-   * [[0, 0, 0, 0],
-   *  [0, 0, 0, 0],
-   *  [0, 0, 0, 0],
-   *  [0, 0, 0, 0]]
-   *
-   * If passed, the board will be initialized with the provided
-   * initial state.
-   */
-  constructor(initialState) {
-    // eslint-disable-next-line no-console
-    console.log(initialState);
+  constructor(
+    initialState = [
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+    ],
+    rows = 4,
+    columns = 4,
+  ) {
+    this.initialState = initialState;
+    this.rows = rows;
+    this.columns = columns;
+    this.score = 0;
+    this.status = 'idle';
   }
 
-  moveLeft() {}
-  moveRight() {}
-  moveUp() {}
-  moveDown() {}
-
-  /**
-   * @returns {number}
-   */
-  getScore() {}
-
-  /**
-   * @returns {number[][]}
-   */
-  getState() {}
-
-  /**
-   * Returns the current game status.
-   *
-   * @returns {string} One of: 'idle', 'playing', 'win', 'lose'
-   *
-   * `idle` - the game has not started yet (the initial state);
-   * `playing` - the game is in progress;
-   * `win` - the game is won;
-   * `lose` - the game is lost
-   */
-  getStatus() {}
-
-  /**
-   * Starts the game.
-   */
-  start() {}
-
-  /**
-   * Resets the game.
-   */
-  restart() {}
-
-  // Add your own methods here
+  updateTile(tile, num) {
+    tile.innerText = num > 0 ? num : '';
+    tile.classList.value = '';
+    tile.classList.add('tile');
+
+    if (num > 0) {
+      tile.classList.add('--' + num.toString());
+    }
+  }
+
+  filterZero(row) {
+    return row.filter((num) => num !== 0);
+  }
+
+  slide(row) {
+    let newRow = this.filterZero(row);
+
+    for (let i = 0; i < newRow.length - 1; i++) {
+      if (newRow[i] === newRow[i + 1]) {
+        newRow[i] *= 2;
+        newRow[i + 1] = 0;
+        this.score += newRow[i];
+      }
+    }
+
+    newRow = this.filterZero(newRow);
+
+    while (newRow.length < this.columns) {
+      newRow.push(0);
+    }
+
+    return newRow;
+  }
+
+  setupMovements() {
+    document.addEventListener('keyup', (e) => {
```

Comment:
Adding event listeners directly in a class method can lead to memory leaks if not removed properly. Ensure that listeners are removed when the game object is no longer needed or when the game ends.